### PR TITLE
Binary media types bug fix

### PR DESCRIPTION
**Pull request by keetonian**
*Issue #, if available:*
#1036 

*Description of changes:*
Converts `~1` to `/`, as is expected in the swagger for binary media types.

The CloudFormation property expects `~1`. Docs: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-restapi.html#cfn-apigateway-restapi-binarymediatypes

The Swagger property expects `/`. Docs: https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-swagger-extensions-binary-media-types.html

*Description of how you validated changes:*
Tests updated and pass.

*Checklist:*

- [x] Write/update tests
- [x] `make pr` passes
- [x] Update documentation
- [x] Verify transformed template deploys and application functions as expected

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samtranslator/swagger/swagger.py [hunk 1]
```diff
@@ -257,7 +258,8 @@ def add_cors(self, path, allowed_origins, allowed_headers=None, allowed_methods=
                                                                                            allow_credentials)
 
     def add_binary_media_types(self, binary_media_types):
-        self._doc[self._X_APIGW_BINARY_MEDIA_TYPES] = binary_media_types
+        bmt = json.loads(json.dumps(binary_media_types).replace('~1', '/'))
```
Comment (keetonian) [inline samtranslator/swagger/swagger.py]: Dump it into json so all nested lists and dictionaries are expressed as a string, replace the `~1` with `/`, and then load back into data form.

File: tests/translator/input/api_with_binary_media_types.yaml [hunk 2]
```diff
@@ -1,8 +1,12 @@
+Parameters:
+  BMT:
+    Type: String
+    Default: 'image~1/jpg'
```
Comment (jlhood) [inline tests/translator/input/api_with_binary_media_types.yaml]: Why `~1/`?

Comment (keetonian) [reply tests/translator/input/api_with_binary_media_types.yaml]: I thought I'd fixed that. One sec, will push from local.

File: samtranslator/model/sam_resources.py [hunk 3]
```diff
@@ -475,6 +475,9 @@ def to_cloudformation(self, **kwargs):
         """
         resources = []
 
+        intrinsics_resolver = kwargs["intrinsics_resolver"]
```
Comment (praneetap) [inline samtranslator/model/sam_resources.py]: Do we have full test coverage on the different intrinsics supported?

Comment (keetonian) [reply samtranslator/model/sam_resources.py]: The intrinsics I'm resolving (see next line) are just parameter references, which I included in the tests.

**Pull request merged**